exporter/collector/metrics.go

```diff
@@ -64,14 +71,22 @@ type selfObservability struct {
 
 // MetricsExporter is the GCM exporter that uses pdata directly.
 type MetricsExporter struct {
-	// A channel that receives metric descriptor and sends them to GCM once
-	metricDescriptorC chan *monitoringpb.CreateMetricDescriptorRequest
-	client            *monitoring.MetricClient
-	obs               selfObservability
+	// write ahead log handles exporter retries in-order to handle network outages
+	wal    *wal.Log
+	client *monitoring.MetricClient
+	obs    selfObservability
 	// shutdownC is a channel for signaling a graceful shutdown
 	shutdownC chan struct{}
 	// mdCache tracks the metric descriptors that have already been sent to GCM
 	mdCache map[string]*monitoringpb.CreateMetricDescriptorRequest
+	// A channel that receives metric descriptor and sends them to GCM once
+	metricDescriptorC chan *monitoringpb.CreateMetricDescriptorRequest
+	// the "write" index for WAL
+	wWALIndex *atomic.Uint64
+	// the "read" index for WAL
+	rWALIndex *atomic.Uint64
```

Comment:
Does this actually need to be atomic, or could it be a local variable in the read and export loop? It seems like it's only updated in that one goroutine.

Reply:
removed these